src/Cake.Core.Tests/Unit/CakeEngineTests.cs

```diff
@@ -683,7 +683,7 @@ public void Should_Run_Task_Setup_Before_Each_Task()
                 var result = new List<string>();
                 var fixture = new CakeEngineFixture();
                 var engine = fixture.CreateEngine();
-                engine.RegisterTaskSetupAction((cc, sc) => result.Add("TASK_SETUP:" + sc.Task.Name));
+                engine.RegisterTaskSetupAction(context => result.Add("TASK_SETUP:" + ((ITaskSetupContext)context).Task.Name));
```

Comment:
No, not any more. Originally I did, then I changed it based on your feedback. Will correct.